api/swagger.yaml

```diff
@@ -77,43 +90,337 @@ tags:
+      entity at a time.
+
+      #### Applicable life cycle phases
+
+      | Planned | Projected | Actual | Realized |
+      | :-----: | :-------: | :----: | :------: |
+      |         |           | ●      | ●        |
+
+      ### `locationUpdateEvent`
+
+      Updates the location of a `Vehicle`
+
+      #### Applicable entities
+
+      | Location | Vehicle | Shipment | Trip  | Route | Actor | Constraint | Sensor | Event |
+      | :------: | :-----: | :------: | :---: | :---: | :---: | :--------: | :----: | :---: |
+      |          | ●       |          |       |       |       |            |        |       |
+
+      ### `sensorUpdateEvent`
+
+      Event to send an updated value of a sensor of the `Vehicle`. Can be used e.g. to monitor temperature inside a trailer.
+
+      #### Applicable entities
+
+      | Location | Vehicle | Shipment | Trip  | Route | Actor | Constraint | Sensor | Event |
+      | :------: | :-----: | :------: | :---: | :---: | :---: | :--------: | :----: | :---: |
+      |          |         |          |       |       |       |            | ●      |       |
+
+      ### `startEngineEvent`, `stopEngineEvent`
+
+      The engine of a `Vehicle` is started or stopped.
+
+      #### Applicable entities
+
+      | Location | Vehicle | Shipment | Trip  | Route | Actor | Constraint | Sensor | Event |
+      | :------: | :-----: | :------: | :---: | :---: | :---: | :--------: | :----: | :---: |
+      |          | ●       |          |       |       |       |            |        |       |
+
+      ## Warning events
+
+      Warning events warn about violations, possible causes for delay or other things that may have impact on an operation and may need some
+      attention. Warning events can exist in all life cycle phases, since other events might be the cause of a warning.
+
+      #### Applicable life cycle phases
+
+      | Planned | Projected | Actual | Realized |
+      | :-----: | :-------: | :----: | :------: |
+      |  ●      | ●         | ●      | ●        |
+
+      ### `restrictionWarningEvent`
+
+      A warning event that may be sent when a restriction is violated.
+
+      #### Applicable entities
+
+      | Location | Vehicle | Shipment | Trip  | Route | Actor | Constraint | Sensor | Event |
+      | :------: | :-----: | :------: | :---: | :---: | :---: | :--------: | :----: | :---: |
+      | ●        |         |          | ●     | ●     |       |            |        |       |
+
+      ### `constraintViolationEvent`
+
+      Can be generated if a constraint is violated.
+
+      #### Applicable entities
+
+      | Location | Vehicle | Shipment | Trip  | Route | Actor | Constraint | Sensor | Event |
+      | :------: | :-----: | :------: | :---: | :---: | :---: | :--------: | :----: | :---: |
+      | ●        |         | ●        | ●     |       |       |            | ●      |       |
```

Comment:
Could we give an example for each entity type of what kind of constraints could be violated?

Reply:
Should the constraint entity also be included in the constraint violation event to specify which constraint is violated?